station-world-events-v0/src/main/java/net/modificationstation/stationapi/api/celestial/CelestialEvent.java

```diff
@@ -0,0 +1,270 @@
+                activityState.markDirty();
+                return false;
+            }
+        }
+        for (CelestialEvent otherEvent : dependencies) {
+            if (otherEvent == null) continue;
+            if (!otherEvent.isActive()) {
+                activityState.active = false;
+                activityState.attemptedActivation = true;
+                activityState.markDirty();
+                return false;
+            }
+        }
+        long days = (worldTime / dayLength) + dayOffset;
+        if (days % frequency > extraDays) {
+            activityState.attemptedActivation = false;
+            return false;
+        }
+        if (!activityState.attemptedActivation) {
+            active = days % frequency <= extraDays && random.nextFloat() <= chance;
+        }
+        if (active) {
+            activityState.active = true;
+            activityState.attemptedActivation = true;
+            activityState.markDirty();
+            onActivation();
+        }
+        return active;
+    }
+
+    /**
+     * Called precisely on event activation.
+     * Can be customized by inheriting from this class.
+     * Has access to the world object for more versatility.
+     */
+    public void onActivation() {
+    }
+
+    /**
+     * Called precisely on event deactivation.
+     * Can be customized by inheriting from this class.
+     * Has access to the world object for more versatility.
+     */
+    public void onDeactivation() {
+    }
+
+    /**
+     * Updates the activity status of an already active event.
+     * Used by CelestialTimeManager 4 times per day.
+     * Checks for validity of the time interval but not for chance and incompatibility.
+     *
+     * @param worldTime Current time in the world measured in ticks.
+     */
+    public void updateEvent(long worldTime) {
+        CelestialEventActivityState activityState = (CelestialEventActivityState) this.world.getOrCreateState(CelestialEventActivityState.class, this.name);
+        if (initializationNeeded) {
+            activityState = initializeEvent(activityState);
+        }
+        if (!active && !activityState.active) return;
+        worldTime -= startingDaytime;
+        worldTime += endingDaytime;
+        long days = (worldTime / dayLength) + dayOffset;
+        active = days % frequency <= extraDays;
+        activityState.active = active;
+        if (!active) {
+            onDeactivation();
+            activityState.attemptedActivation = false;
+        }
+        activityState.markDirty();
+    }
+
+    /**
+     * Initializes the event by synchronizing it with NBT data.
+     * Only used internally.
+     *
+     * @param activityState CelestialEventActivityState (custom PersistentState) used for accessing NBT data.
+     * @return The initialized CelestialEventActivityState.
+     */
+    private CelestialEventActivityState initializeEvent(CelestialEventActivityState activityState) {
```

Comment:
`activityState` is re-assigned within the method, making the actual value passed into the method unused.

Reply:
You are correct, the parameter is not needed. The IDE is warning me about this as well

Reply:
The fix has been pushed